DigitalLearningSolutions.Web/ViewModels/TrackingSystem/Delegates/AllDelegatesViewModel.cs

```diff
@@ -0,0 +1,37 @@
+﻿namespace DigitalLearningSolutions.Web.ViewModels.TrackingSystem.Delegates
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using DigitalLearningSolutions.Data.Models.User;
+    using DigitalLearningSolutions.Web.Helpers;
+
+    public class AllDelegatesViewModel
+    {
+        public AllDelegatesViewModel(
+            int centreId,
+            IEnumerable<DelegateUserCard> delegateUsers,
+            CustomPromptHelper helper

```

Comment:
Hmm.. I think passing this helper that fetches data into the ViewModel is a bit of an antipattern. I wouldn't expect ViewModels to be responsible for anything other than being containers for data (and maybe doing some formatting logic or mapping)

I wonder if we could do the mappings to SearchableDelegateViewModel a layer above and pass that into the ViewModel. It might also make sense to add an additional method to CustomPromptHelper that only takes a centreId and a delegate user. :)